amundsen_application/static/js/ducks/tableMetadata/types.ts

```diff
@@ -15,9 +15,11 @@ export enum GetTableData {
 };
 export interface GetTableDataRequest {
   type: GetTableData.REQUEST;
-  key: string;
-  searchIndex?: string;
-  source?: string;
+  payload: {
+    key: string;
+    searchIndex?: string;
+    source?: string;
```

Comment:
If you use default values for the `getTableData` you should be able to make these fields required.

Reply:
Will look into it.